WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -432,6 +458,23 @@ extension ProductFormViewModel {
         }
     }
 
+    func saveProductImagesWhenAllAreUploaded() {
+        productImagesUploader.saveProductImagesWhenNoneIsPendingUploadAnymore(siteID: product.siteID,
+                                                                              productID: product.productID,
+                                                                              isLocalID: !product.existsRemotely) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let images):
+                let currentProduct = self.product
+                self.originalProduct = .init(product: self.originalProduct.product.copy(images: images))
```

Comment:
Should we use `private func resetProduct(_ product: EditableProductModel)` instead of directly setting `originalProduct` here?

Reply:
pro suggestion 💯 ! updated in https://github.com/woocommerce/woocommerce-ios/pull/7048/commits/9dd3efde0d2f70d710601cd8abf7ea90f30a8685